runtime/util/sendslot.c

```diff
@@ -22,29 +22,36 @@
 
 #include "j9.h"
 #include "util_internal.h"
+#include <stdint.h>
 
 /* there is no error checking: the signature MUST be well-formed */
 UDATA
 getSendSlotsFromSignature(const U_8* signature)
 {
 	UDATA sendArgs = 0;
 	UDATA i = 1; /* 1 to skip the opening '(' */
+	UDATA oldI = 0;
 
-	for (; ; i++) {
+	/* All UTF8 in the class file have a size represented by uint16_t. The size check
+	 * is only necessary for ASGCT where the signature provided may not be valid,
+	 * but is harmless in the normal runtime case.
+	 */
+	for (; (i <= UINT16_MAX) && (i > oldI); i++) {
```

Comment:
If always `i <= UINT16_MAX` then I don't see why `oldI` is needed.

Reply:
It's to prevent a possible overrun of the counter (which we did see). It may not be strictly necessary since the MAX check is in each loop.

Reply:
It's worth noting that this is not performance-critical code.

Reply:
Even if it's supposedly not perf critical, the `oldI` check doesn't serve any purpose since 'i <= UINT16_MAX`.

Reply:
oldI removed.